### DISCUSSION TITLE
IBKR IdealPro Paper trading orders not submitting

### DISCUSSION THREAD

**Discussion by matheosz1** [nautechsystems/nautilus_trader#1267] [Q&A] [answered]
I am trying to paper trade to ensure my ordering is correct. The bar type is MID-EXTERNAL and I am making market orders. This is the error I keep getting: 
<img width="807" alt="Screenshot 2023-10-02 at 2 07 26 PM" src="https://github.com/nautechsystems/nautilus_trader/assets/104100249/dc864761-bf86-45be-8a7f-dcdf35d7b1f5">

Does anyone have tips in overcoming this? 

I am using this configuration:
`
ib_contracts = [
    IBContract(
        secType="CASH",
        symbol="EUR",
        exchange="IDEALPRO",
        currency="USD"
    ),
    IBContract(
        secType="CASH",
        symbol="GBP",
        exchange="IDEALPRO",
        currency="USD"
    ),
    IBContract(
        secType="CASH",
        symbol="AUD", 
        exchange="IDEALPRO", 
        currency="USD"),
]

gateway = InteractiveBrokersGatewayConfig(
start=False,
username=os.environ['IB_ACCOUNT'],
password=os.environ['IB_PASSWORD'],
trading_mode=os.environ['TRADE_MODE'],
read_only_api=True,
)

instrument_provider = InteractiveBrokersInstrumentProviderConfig(
    load_ids=frozenset(
        [
            "EUR/USD.IDEALPRO",
            "GBP/USD.IDEALPRO",
            "AUD/USD.IDEALPRO",
        ],
    ),
    load_contracts=frozenset(ib_contracts),

)

config_node = TradingNodeConfig(
    trader_id="TESTER-001",
    logging=LoggingConfig(log_level="INFO"),
    risk_engine=LiveRiskEngineConfig(bypass=True),
    data_clients={
        "IB": InteractiveBrokersDataClientConfig(
            ibg_host="127.0.0.1",
            ibg_port=7497,
            ibg_client_id=1,
            handle_revised_bars=False,
            use_regular_trading_hours=True,
            instrument_provider=instrument_provider,
            gateway=gateway,
        ),
    },
    exec_clients={
        "IB": InteractiveBrokersExecClientConfig(
            ibg_host="127.0.0.1",
            ibg_port=7497,
            ibg_client_id=1,
            account_id=os.environ['ACCOUNT_ID'], 
            gateway=gateway,
            instrument_provider=instrument_provider,
            routing=RoutingConfig(default=True, venues=frozenset({"IDEALPRO"})),
        ),
    },
    data_engine=LiveDataEngineConfig(
        time_bars_timestamp_on_close=False, 
        validate_data_sequence=True,  
    ),
    timeout_connection=90.0,
    timeout_reconciliation=5.0,
    timeout_portfolio=5.0,
    timeout_disconnection=5.0,
    timeout_post_stop=2.0,
) 
`
Thank you in advance.

**Comment by cjdsellers** [comment]
@rsmb7z any thoughts on this?

**Reply by rsmb7z** [reply, marked as answer]
@matheosz1 Most likely it due to immediate rejection by TWS without any update hence being Order being canceled on Nautilus as well. If you enable the logs in DEBUG mode, will find exact reason.

Btw you don't need to use `ib_contracts` which is redundant in your case and is more advanced optional alternative for loading instruments.

**Reply by matheosz1** [reply]
Hey @rsmb7z, thank you for the tip. It turns out I needed to round to the nearest dollar. Perhaps there's a setting which allows me to send orders with cents however I do not need it.

I will use debug logs when needed from here on out so thanks again!
